src/Backend/Modules/Pages/Domain/ModuleExtra/ModuleExtra.php

```diff
@@ -73,15 +73,15 @@ public function __construct(
         string $type,
         string $label,
         ?string $action,
-        ?string $data,
+        $data,
         bool $hidden,
         int $sequence
     ) {
         $this->module = $module;
         $this->type = $type;
         $this->label = $label;
         $this->action = $action;
-        $this->data = $data;
+        $this->data = serialize($data);
```

Comment:
I would keep the data as is and use a Life Cycle Event
Now if you want to do something with `$data` you will always need to unserialise it first